src/services/new_mint.rs

```diff
@@ -89,10 +98,10 @@ async fn get_conditional_vault(
     Ok(vault)
 }
 
-fn get_relevant_accounts_from_mint_and_vault(
-    mint_instruction: &crate::entities::transactions::Instruction,
-    conditional_vault: ConditionalVault,
-) -> Vec<(&str, String)> {
+fn get_relevant_accounts_from_mint_and_vault<'a>(
```

Comment:
did we need to add lifetimes here because now we are doing a reference to ConditionalVault?

Reply:
We need to use lifetime for the `Instruction` because we are returning borrowed value from itin the Vec `Vec<(&'a str, String)>` which is `&str`. So here we make sure that `&str` inside `Vec` lives as long as `Instruction`.

Reply:
ok sweet sounds good the. And once that is cleaned up then I'll approve!